pwiz_tools/Shared/Common/UserInterfaces/IUserInterface.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Threading;
+using System.Windows.Forms;
+using JetBrains.Annotations;
+using pwiz.Common.SystemUtil;
+
+namespace pwiz.Common.UserInterfaces
+{
+    public interface IUserInterface
+    {
+        void DisplayMessage(Control owner, string message);
```

Comment:
You need to get rid of the Windows.Forms use in the interface at least. That means no Control and now MessageBoxButtons in the interface. I would expect the owner control to be passed in at construction time of the object. It just knows the owner control and correctly displays the message box when asked to show a message.

DisplayMessageWithButtons() would need to become DisplayMessageWithOptions() and the Windows.Forms implementation would need to translate between the options and buttons, while the text version would ignore them? That worries me a little, since they seem likely to be questions expecting an answer.